apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -1045,6 +1173,30 @@ func (s *StorageProvisionerAPIv3) oneVolumeBlockDevice(
 	if err != nil {
 		return state.BlockDeviceInfo{}, err
 	}
+	// TODO(gsamfira): revisit this
+	planCanAccess, err := s.getMachineAuthFunc()
+	if err != nil && !errors.IsNotFound(err) {
+		return state.BlockDeviceInfo{}, err
+	}
+	var blockDeviceInfo state.BlockDeviceInfo
+
+	volumeAttachmentPlan, err := s.oneVolumeAttachmentPlan(id, planCanAccess)
+	if err != nil {
+		// This is optional. It should not err out
```

Comment:
Being clearer about what "this" is is probably useful.
Is it the BlockDeviceInfo that is optional?

Reply:
Will update to be more clear. It refers to the volume attachment plans. They are not mandatory. If missing, defaults to old code path.